app/lib/dartdoc/dartdoc_runner.dart

```diff
@@ -40,6 +40,56 @@ class DartdocJobProcessor extends JobProcessor {
   DartdocJobProcessor({Duration lockDuration})
       : super(service: JobService.dartdoc, lockDuration: lockDuration);
 
+  /// Uses the tool environment's SDK (the one that is used for analysis too) to
+  /// generate dartdoc documentation and extracted data file for SDK API indexing.
+  /// Only the extracted data file will be used and uploaded.
+  Future generateDocsForSdk() async {
+    if (await dartdocBackend.hasValidDartSdkDartdocData()) return;
+    final tempDir =
+        await Directory.systemTemp.createTemp('pub-dartlang-dartdoc');
+    try {
+      final tempDirPath = tempDir.resolveSymbolicLinksSync();
+      final outputDir = tempDirPath;
+      final args = [
+        '--sdk-docs',
+        '--output',
+        outputDir,
+        '--hosted-url',
+        siteRoot,
+        '--link-to-remote',
+        '--no-validate-links',
+      ];
+      if (envConfig.toolEnvDartSdkDir != null) {
+        args.addAll(['--sdk-dir', envConfig.toolEnvDartSdkDir]);
+      }
+      final pr = await runProc(
+        'dart',
+        ['bin/pub_dartdoc.dart']..addAll(args),
+        workingDirectory: _pkgPubDartdocDir,
+        timeout: _dartdocTimeout * 2,
+      );
+
+      final pubDataFile = new File(p.join(outputDir, 'pub-data.json'));
+      final hasPubData = await pubDataFile.exists();
+      final isOk = pr.exitCode == 0 && hasPubData;
+      if (!isOk) {
+        _logger.warning(
+            'Error while generating SDK docs.\n\n${pr.stdout}\n\n${pr.stderr}');
```

Comment:
Should this be an error? Can this be triggered by poor user code? In that case, can we recognize the difference between bad user input and an infrastructure problem?

Reply:
It is most likely an infrastructure problem, making it a hard error.